pkg/grafana/client/credential.go

```diff
@@ -0,0 +1,38 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type adminCredentials struct {
+	Username string
+	Password string
+}
+
+func getAdminCredentials(ctx context.Context, client client.Client) (adminCredentials, error) {
+	grafanaAdminSecret := &v1.Secret{}
+	err := client.Get(ctx, types.NamespacedName{
+		Namespace: Namespace,
+		Name:      GrafanaSecretName,
+	}, grafanaAdminSecret)
+	if err != nil {
+		return adminCredentials{}, errors.WithStack(err)
+	}
+
+	adminUser, ok := grafanaAdminSecret.Data["admin-user"]
```

Comment:
Maybe we should use the present approach like in tls and just have 1 error that says the credentials is incorrect.

Reply:
Done